packages/ses/src-xs/index.js

```diff
@@ -0,0 +1,463 @@
+
+    if (importHook) {
+      /** @param {string} specifier */
+      const nativeImportHook = async specifier => {
+        await null;
+        let descriptor =
+          mapGet(descriptors, specifier) ??
+          moduleMapHook(specifier) ??
+          (await importHook(specifier));
+        mapDelete(descriptors, specifier);
+        descriptor = adaptModuleDescriptor(descriptor, specifier, compartment);
+        return descriptor;
+      };
+      nativeOptions = {
+        ...nativeOptions,
+        resolveHook,
+        importHook: nativeImportHook,
+        loadHook: nativeImportHook,
+      };
+    }
+
+    if (importNowHook) {
+      /** @param {string} specifier */
+      const nativeImportNowHook = specifier => {
+        let descriptor =
+          mapGet(descriptors, specifier) ??
+          moduleMapHook(specifier) ??
+          importNowHook(specifier);
+        mapDelete(descriptors, specifier);
+        descriptor = adaptModuleDescriptor(descriptor, specifier, compartment);
+        return descriptor;
+      };
+      nativeOptions = {
+        ...nativeOptions,
+        resolveHook,
+        importNowHook: nativeImportNowHook,
+        loadNowHook: nativeImportNowHook,
+      };
+    }
+
+    const compartment = new NativeCompartment(nativeOptions);
+
+    const nativeGlobalThis = nativeGetGlobalThis(compartment);
+
+    const nativeEval = nativeGlobalThis.eval;
+
+    weakmapSet(privateFields, compartment, {
+      name,
+      transforms,
+      delegateNative,
+      noNamespaceBox,
+      nativeEval,
+      descriptors,
+    });
+
+    const shimOptions = {
+      ...options,
+      __noNamespaceBox__: true,
+      __options__: true,
+    };
+
+    uncurryThis(ShimCompartment)(compartment, shimOptions);
+
+    const shimGlobalThis = shimGetGlobalThis(compartment);
+
+    const ChildCompartment = nativeEvaluate(compartment, compartmentShim)(
+      compartmentShim,
+      commons,
+      nativeGlobalThis.Compartment,
+      shimGlobalThis.Compartment,
+      maybeHarden,
+    );
+
+    defineProperty(nativeGlobalThis, 'Compartment', {
+      value: ChildCompartment,
+      writable: true,
+      configurable: true,
+      enumerable: false,
+    });
+
+    defineProperty(shimGlobalThis, 'Compartment', {
+      value: ChildCompartment,
+      writable: true,
+      configurable: true,
+      enumerable: false,
+    });
+
+    maybeHarden(compartment);
+    return compartment;
+  }
+
+  return maybeHarden(Compartment);
+};`;
+
+// Adapt the start compartment's native Compartment to the SES-compatibility
+// adapter.
+// Before Lockdown, the Compartment constructor in transitive child
+// Compartments is not (and cannot be) hardened.
+const noHarden = object => object;
+globalThis.Compartment = FERAL_EVAL(compartmentShim)(
+  compartmentShim,
+  commons,
+  NativeCompartment,
+  ShimCompartment,
+  noHarden,
+);
+
+globalThis.lockdown = options => {
+  const hardenIntrinsics = repairIntrinsics(options);
```

Comment:
I notice that this PR does not modify `lockdown.js` which defines the `repairIntrinsics` being called here. Nor does it modify `tame-harden.js` or `make-hardener.js` used by `repairIntrinsics` to make the `harden` function. But presumably, on XS we want to use only the XS native `harden`. How does that happen?

Reply:
We use native harden due to an existing fall through to globalThis.harden in `makeHarden`. We also continue to respect the lockdown option to make `harden` a no-op. I’ve elected to continue using our `lockdown` variant so-far, since it is parameterized, can conduct repairs to the XS intrinsics if any become necessary, and support enablements. I’m at least leaving native lockdown out of scope for this run at native Compartment and ModuleSource.